datafusion/core/src/datasource/physical_plan/parquet/row_groups.rs

```diff
@@ -1105,13 +1100,18 @@ mod tests {
         let data = bytes::Bytes::from(std::fs::read(path).unwrap());
 
         // generate pruning predicate
-        let schema = Schema::new(vec![
-            Field::new("String", DataType::Utf8, false),
-            Field::new("String3", DataType::Utf8, false),
-        ]);
-        let sql =
-            "SELECT * FROM tbl WHERE \"String\" IN ('Hello_Not_Exists', 'Hello_Not_Exists2')";
-        let expr = sql_to_physical_plan(sql).unwrap();
```

Comment:
This previous test relied on the optimizer who will convert `in_list` expr with no more 3 elements to `and` expr. Now we don't need this. `logical2physical` would convert `logical_expr` to `physical_expr` without optimizer.

Reply:
I agree -- this change makes sense

BTW I tried to improve these tests to avoid duplication in https://github.com/apache/arrow-datafusion/pull/8435 (not yet merged). I'll try and consolidate this test too when I get a chance